testsuite/MDAnalysisTests/core/test_topologyattrs.py

```diff
@@ -60,16 +61,26 @@ class TopologyAttrMixin(object):
     Ridx = np.array([0, 0, 2, 2, 1, 1, 3, 3, 1, 2])
     Sidx = np.array([0, 1, 1, 0])
 
+    # single_idx = np.array([0])
+
     def setUp(self):
         self.top = Topology(10, 4, 2,
                             attrs=[self.attrclass(self.values.copy())],
                             atom_resindex=self.Ridx,
                             residue_segindex=self.Sidx)
-
         self.attr = getattr(self.top, self.attrclass.attrname)
 
+        # TODO: should we treat single element attr specially. We do that
```

Comment:
Code in Masses
```python
    def get_residues(self, rg):
        resatoms = self.top.tt.residues2atoms_2d(rg._ix)

        if isinstance(rg._ix, int):
            # for a single residue
            masses = self.values[resatoms].sum()
        else:
            # for a residuegroup
            masses = np.empty(len(rg))
            for i, row in enumerate(resatoms):
                masses[i] = self.values[row].sum()

        return masses
```

code in Charges
```python
    def get_residues(self, rg):
        charges = np.empty(len(rg))

        resatoms = self.top.tt.residues2atoms_2d(rg._ix)

        for i, row in enumerate(resatoms):
            charges[i] = self.values[row].sum()

        return charges

```

For masses we return a different type if `rg` only contains one element. I'm also not sure if `_ix` can be an int at runtime.